### PR TITLE
Add constructors and operators

### PR DESCRIPTION
**Pull request by maciejewiczow**
Added constructors and assigmnent operators according to the rule of five to all other classes that needed it (according to the appropriate Unload*() functions found in raylib). 

Also some cosmetic fixes:
* Replaced all uses of `NULL` with `nullptr`
* Fixed a memory leak in `ModelAnimation::Load` method

### CODE REVIEW

File: include/Material.hpp [hunk 1]
```diff
@@ -33,6 +42,7 @@ class Material : public ::Material {
      */
     static std::vector<Material> Load(const std::string& fileName) {
         int count = 0;
+        // this function possibly leaks the materials array
```
Comment (RobLoach) [inline include/Material.hpp:45]:
``` suggestion
        // TODO: Material::Load() possibly leaks the materials array.
```

**Comment by RobLoach** [comment]
This is absolutely fantastic! Thanks so much for pushing it forwards. Is there anything else you think we should get in there before merging?

**Comment by maciejewiczow** [comment]
Idk, nothing really comes to my mind right now. 
There is one thing about the `const` modifier in various class methods. There are more places that it could be added in, but then these methods would need to be duplicated, for const and non-const version. 
Without the duplication const and non-const methods could not be mixed in chain calls. But the duplication process can be quite laborious and I did not include it here.

**Comment by maciejewiczow** [comment]
Also as a general note, not necessarily related to this PR, it would be neat if all the methods that take `std::string` would also provide a `const char*` alternative, as the string class always performs a copy when you pass in a `const char*`, even when the argument is declared as a const reference. So the char pointer should be preferred unless you really need some of the string class functionality, because conversion from `char*` to `string` is more taxing than from `string` to `char*`.

**Comment by maciejewiczow** [comment]
Also have you maybe considered using some framework for the tests?

**Comment by RobLoach** [comment]
> So the char pointer should be preferred unless you really need some of the string class functionality, because conversion from char* to string is more taxing than from string to char*.

Having `const char*` overrides seems like a good idea. I've heard some things on `string_view`. Would that be worth considering?

**Comment by maciejewiczow** [comment]
> Having `const char*` overrides seems like a good idea. I've heard some things on `string_view`. Would that be worth considering?

I don't think string view would be a good match here. String view is good for efficiently referencing parts of other, existing string. I think plain old `const char*` will be good fit here.

**Comment by maciejewiczow** [comment]
> Having `const char*` overrides seems like a good idea. I've heard some things on `string_view`. Would that be worth considering?

I don't think that string view would be a good fit here. String view is good at efficiently referencing a part of other, already existing string. Plain old `const char*` will be good here.

**Pull request merged**